tensorboard/components/tf_runs_selector/tf-wbr-string.html

```diff
@@ -0,0 +1,75 @@
+<!--
+@license
+Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="../tf-imports/polymer.html" />
+
+<!--
+  tf-wbr-string safely renders a string, with <wbr> elements inserted
+  around select delimiters.
+-->
+<dom-module id="tf-wbr-string">
+  <template>
+    <!--
+      This ugly formatting is required to prevent spaces from slipping
```

Comment:
Optional but IMO it can be nicer to achieve this by inserting inline comments between the tags, like

```html
<template is="dom-repeat" items="[[_parts]]" as="part"><!--
  -->[[part.contents]]<!--
  --><template is="dom-if" if="[[!part.last]]"><!--
    --><wbr /><!--
  --></template><!--
--></template>
```

Then at least you can preserve the hierarchical shape, though obviously it's a bit more verbose and cumbersome to write.

Reply:
Agreed, but Prettier rewrites this to

```html
    <template is="dom-repeat" items="[[_parts]]" as="part"
      ><!--
      -->[[part.contents]]<!--
      --><template is="dom-if" if="[[!part.last]]"
        ><!--
        --><wbr /><!--
      --></template
      ><!--
    --></template>
```

<sub>Daily reminder that JSX does not have semantic whitespace.</sub>